[bitnami/postgresql] Add hostNetwork option for PostgreSQL StatefulSets

hjames9 opened the pull request:
Signed-off-by: Hayden James <hayden.james@gmail.com>

This adds the "hostNetwork" option to the StatefulSets used for the PostgreSQL chart in order to enable/disable that feature.  This allows you to enable host networking for the PostgreSQL pod, instead of only forcing the user to user a network bridge.

**Checklist**
<!-- [Place an '[X]' (no spaces) in all applicable fields. Please remove unrelated fields.] -->
- [X] Chart version bumped in `Chart.yaml` according to [semver](http://semver.org/).
- [X] Variables are documented in the values.yaml and added to the README.md using (readme-generator-for-helm)[https://github.com/bitnami-labs/readme-generator-for-helm]
- [X] Title of the PR starts with chart name (e.g. [bitnami/<name_of_the_chart>])
- [X] All commits signed off and in agreement of [Developer Certificate of Origin (DCO)](https://github.com/bitnami/charts/blob/master/CONTRIBUTING.md#sign-your-work)

carrodher commented:
Thanks for your contribution, please note that in order to make the Pull Requests' (PRs) testing and merging process easier, you need to submit changes to multiple charts in separate PRs. Can you split this PR into 2 different ones?

javsalgar commented:
Could you merge from master? We recently added a PR and found some issues. Sorry for the inconvenience

hjames9 commented:
@carrodher Thanks for the feedback.  I've updated this PR to only modify the bitnami/postgresql chart.  And I've opened a separate PR for the bitnami/postgresql-ha chart. https://github.com/bitnami/charts/pull/9071

@javsalgar I've also rebased on master

marcosbc reviewed (changes requested):
Thanks for the contribution! A small request, could you also add the entry to the `values.yaml` file of the PostgreSQL chart? Note that the README table generation is automated and if it is not added, those changes could be lost.

hjames9 commented:
@marcosbc Disregard last message.  I fixed the issue and updated the PR.  It should be good now.

hjames9 commented:
@juan131 Could you review this one also?  Same as my other PRs.  Thanks!

marcosbc reviewed (approved):
LGTM!
